src/sardana/macroserver/msmacromanager.py

```diff
@@ -1290,9 +1294,31 @@ def __runXMLMacro(self, xml):
     _runXMLMacro = __runXMLMacro
 
     def runMacro(self, macro_obj):
+
         name = macro_obj._getName()
         desc = macro_obj._getDescription()
         door = self.door
+        self.macro_obj = macro_obj
+
+        if macro_obj.getParentMacro() is None:
+            try:
+                self.logging_onoff = macro_obj.getEnv("LogMacroOnOff")
+            except:
+                self.logging_onoff = False
+            if self.logging_onoff:
+                try:
+                    self.logging_path = macro_obj.getEnv("LogMacroPath")
+                except:
+                    macro_obj.setEnv("LogMacroPath", "/tmp")
+                    self.logging_path = macro_obj.getEnv("LogMacroPath")
+                log_file = self.logging_path + "/spock_session.log"
```

Comment:
I would not hard code the file name. Maybe `ScanFile` variable could be used?

Reply:
Our users wanted hard coded names. In the last version they have we had to append to the logfile name the name of the node where spock was running. They wanted like this. If you agree, I could put the hardcoded name in case LogMacroFile is not defined. Is this fine for you?

Reply:
I agree.